core/vm/actor/builtin/types/miner/v2/monies.cpp

```diff
@@ -0,0 +1,187 @@
+
+    const ChainEpoch relevant_replaced_age =
+        std::min(replaced_sector_age, lifetime_cap - capped_sector_age);
+    expected_reward =
+        expected_reward + replaced_day_reward * relevant_replaced_age;
+
+    const TokenAmount penalized_reward =
+        expected_reward * termination_reward_factor.numerator;
+
+    OUTCOME_TRY(penalty,
+                pledgePenaltyForTerminationLowerBound(
+                    reward_estimate, network_power_estimate, sector_power));
+
+    return std::max(
+        penalty,
+        BigInt(BigInt(twenty_day_reward_activation)
+               + bigdiv(BigInt(penalized_reward),
+                        (BigInt(kEpochsInDay))
+                            * termination_reward_factor.denominator)));
+  }
+
+  outcome::result<TokenAmount> Monies::preCommitDepositForPower(
+      const FilterEstimate &reward_estimate,
+      FilterEstimate network_power_estimate,
+      const StoragePower &sector_power) {
+    return expectedRewardForPower(reward_estimate,
+                                  network_power_estimate,
+                                  sector_power,
+                                  precommit_deposit_projection_period);
+  }
+
+  outcome::result<TokenAmount> Monies::initialPledgeForPower(
+      const StoragePower &power,
+      const StoragePower &baseline_power,
+      const TokenAmount &network_total_pledge,
+      const FilterEstimate &reward_estimate,
+      const FilterEstimate &network_power_estimate,
+      const TokenAmount &network_circulation_supply_smoothed) {
+    OUTCOME_TRY(ipBase,
+                expectedRewardForPower(reward_estimate,
+                                       network_power_estimate,
+                                       power,
+                                       initial_pledge_projection_period));
+
+    const BigInt lock_target_num = initial_pledge_lock_target.numerator
+                                   * network_circulation_supply_smoothed;
+    const BigInt lock_target_denom = initial_pledge_lock_target.denominator;
+
+    const StoragePower pledge_share_num = power;
+    const StoragePower network_qa_power = estimate(network_power_estimate);
+    const BigInt pledge_share_denom =
+        std::max(std::max(network_qa_power, baseline_power), power);
+    const BigInt additional_ip_num = lock_target_num * pledge_share_num;
+    const BigInt additional_ip_denom = lock_target_denom * pledge_share_denom;
+    const BigInt additional_ip = bigdiv(additional_ip_num, additional_ip_denom);
+
+    const BigInt nominal_pledge = ipBase + additional_ip;
+    const BigInt space_race_pledge_cap = initial_pledge_max_per_byte * power;
+    return std::min(nominal_pledge, space_race_pledge_cap);
+  }
+
+  outcome::result<TokenAmount> Monies::repayDebtsOrAbort(
+      runtime::Runtime &runtime, Universal<MinerActorState> miner_state) {
+    auto curr_balance = runtime.getCurrentBalance();
+    // TODO(m.tagirov): MinerSate.RepayDebts(curr_balance);
```

Comment:
merged in #451

Reply:
Yes, @Elestrias this method is ready, please use it here